### ISSUE TITLE
[editor] Ctrl+Z does not return the prompts - input however returns the outputs - data

### ISSUE THREAD

**Issue by vanshb03** [lastmile-ai/aiconfig#1457]
### Proposed Solution:

When the user presses Ctrl+Z, the editor should undo the last action, including both the input and output, to restore the previous state compared to solely the output as it is right now. This behavior would be consistent with the interface, where Ctrl+Z typically undoes the last text entry or command. By restoring the input as well, users can easily revert to a previous state for their prompt input.

### Expected Behavior:
- User inputs a command or text in the editor.
- User presses Ctrl+Z.
- The editor undoes the last action, including both the input and output.
- The editor returns to the state before the last action, showing the previous input prompt and removing the output data.
- The user can continue working from the restored state, with the option to redo the undone action if needed.

### Current Behavior:
- User inputs a command or text in the editor.
- The user presses Ctrl+Z when clicking on the input box.
- The editor undoes the last action, which is solely the previous output.

![LASTMILE #1457](https://github.com/lastmile-ai/aiconfig/assets/48507154/c9f72a1e-ed84-4d08-83ed-6e0bce2812c0)

**Comment by vanshb03** [comment]
### Suggested Labels:

- [type: bug](https://github.com/lastmile-ai/aiconfig/labels/type%3A%20bug) [pri: critical](https://github.com/lastmile-ai/aiconfig/labels/pri%3A%20critical) [status: planned](https://github.com/lastmile-ai/aiconfig/labels/status%3A%20planned) [sdk](https://github.com/lastmile-ai/aiconfig/labels/sdk) [effort: mid](https://github.com/lastmile-ai/aiconfig/labels/effort%3A%20mid)

**Comment by saqadri** [comment]
@vanshb03 thanks a lot for reporting, and for your suggestion. @rholinshead what do you think about this? We are currently doing undo/redo just based on the state that is propagated to the TextDocument

**Comment by vanshb03** [comment]
So to recreate the issue, you have to click in the box before you Ctrl+Z, that is when the output box is the only one that changes and the input box remains the same as shown on the issue.

@rholinshead @saqadri

**Comment by vanshb03** [comment]
Yes, I was proposing that Ctrl+Z would only undo the input and NOT the output.

## Suggested Labels:
[status: planned](https://github.com/lastmile-ai/aiconfig/labels/status%3A%20planned) [sdk](https://github.com/lastmile-ai/aiconfig/labels/sdk)
